parts/k8s/cloud-init/artifacts/cse_install.sh

```diff
@@ -169,6 +170,18 @@ installNetworkPlugin() {
     rm -rf $CNI_DOWNLOADS_DIR &
 }
 
+installBcc() {
+    echo "Installing BCC tools..."
+    IOVISOR_KEY_TMP=/tmp/iovisor-release.key
+    IOVISOR_URL=https://repo.iovisor.org/GPG-KEY
+    retrycmd_if_failure_no_stats 120 5 25 curl -fsSL $IOVISOR_URL > $IOVISOR_KEY_TMP || exit $ERR_IOVISOR_KEY_DOWNLOAD_TIMEOUT
+    wait_for_apt_locks
+    retrycmd_if_failure 30 5 30 apt-key add $IOVISOR_KEY_TMP || exit $ERR_IOVISOR_APT_KEY_TIMEOUT
+    echo "deb https://repo.iovisor.org/apt/${UBUNTU_CODENAME} ${UBUNTU_CODENAME} main" > /etc/apt/sources.list.d/iovisor.list
```

Comment:
do we need to do anything special to make this idempotent? If the bcc tools are already installed on the vhd, we'd want to skip these steps, right?

Reply:
apt should just no-op the steps which is desired, I could do a better job cleaning up the files though. I'm not sure it's worth the extra logic to do something like `find /usr/lib -name libbcc.so.0` to detect the installed lib over letting apt handle this itself. L180 I used '>' over '>>' to force the file contents to be fresh/correct.

Reply:
let me know what you prefer?

Reply:
you could just move the call to `installBcc` to inside the `installDeps` function which only gets called for non-VHD deployments

Reply:
I thought that's what `"$FULL_INSTALL_REQUIRED" = "true"` was for, is that not what this curently does? The only reason I didn't put it inside `installDeps` was to keep the additional apt repo separate like the other drivers. I can just nudge it in there though

Reply:
oh yeah where it's at right now is perfectly fine, didn't see you already put it inside the full install clause 👍